### PR TITLE
[iOS] Apply Sponsor View Design

### LINKED ISSUE #558: [iOS] Apply SponsorView design
**Idea Description**

- Title says all

**Reference images and links**
スクリーンショット 2022-09-19 12 50 38

### PR DESCRIPTION
**Pull request by sudachi808**
## Issue
- close #558

## Overview (Required)
- Apply the design.
- Open link with SafariView when items are tapped.
- Show spinner during loading data.
- I feel scrolling is not smooth 🤔 (Images are too large??)

## Links
- 

## Screenshot
PLATINUM-GOLD | GOLD-SUPPORTER | Loading
:--: | :--: | :--:
 |  |

### CODE REVIEW

File: app-ios/Sources/SponsorFeature/SponsorView.swift [hunk 1]
```diff
@@ -21,20 +82,143 @@ public struct SponsorView: View {
...
+    var body: some View {
+        ZStack {
+            AssetColors.white.swiftUIColor
+            Button {
+                action()
+            } label: {
+                NetworkImage(url: URL(string: sponsor.logo))
+                    .scaledToFit()
+                    .frame(minWidth: 0, maxWidth: .infinity)
+                    .frame(minHeight: 0, maxHeight: .infinity)
```
Comment (ry-itto) [inline app-ios/Sources/SponsorFeature/SponsorView.swift:184]: I'm just curious this lines. 
Does specify `.infinity` to both argument isn't working?

```suggestion
                    .frame(minWidth: .infinity, maxWidth: .infinity)
```

File: app-ios/Sources/SponsorFeature/SponsorView.swift [hunk 2]
```diff
@@ -21,20 +82,143 @@ public struct SponsorView: View {
...
+        }
+    }
+}
+
+struct SponsorGridView: View {
+
+    let title: String
+    let sponsors: [Sponsor]
+    let columns: Int
+    var action: (Sponsor) -> Void = { _ in }
```
Comment (ry-itto) [inline app-ios/Sources/SponsorFeature/SponsorView.swift:147]: How about use initializer argument?
My idea is bellow.

```swift
SponsorGridView(
  title: "SPONSORS",
  sponsors: viewStore.sponsors.filter { $0.plan == .supporter },
  columns: 2,
  action: { sponsor in
    viewStore.send(.showLinkSheet(sponsor))
  }
)
```

This way is used in `SwiftUI.Button` so I think this is usual way.

File: app-ios/Sources/SponsorFeature/SponsorView.swift [hunk 3]
```diff
@@ -21,20 +82,143 @@ public struct SponsorView: View {
...
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text(title)
+                .font(.system(size: 22, weight: .semibold, design: .default))
+            LazyVGrid(columns: Array(repeating: GridItem(spacing: 16), count: columns), spacing: 16) {
+                ForEach(sponsors, id: \.self) { sponsor in
```
Comment (ry-itto) [inline app-ios/Sources/SponsorFeature/SponsorView.swift:154]: How about confirm protocol `Identifiable` to `Sponsor` Model with extension?
if it is, `id: \.self` can be omitted 👍🏼

File: app-ios/Sources/SponsorFeature/SponsorView.swift [hunk 4]
```diff
@@ -21,20 +82,143 @@ public struct SponsorView: View {
     }
 
     public var body: some View {
-        Text("TODO: Sponsor")
+        WithViewStore(store) { viewStore in
+            ZStack(alignment: .center) {
+                if viewStore.isLoading {
+                    ActivityIndicator(color: AssetColors.onBackground.swiftUIColor)
```
Comment (woxtu) [inline app-ios/Sources/SponsorFeature/SponsorView.swift:88]: Can we use `ProgressView`?

**Review by sudachi808 — COMMENTED** [review]
fix issues 🙏

File: app-ios/Sources/SponsorFeature/SponsorView.swift (hunk 1 above)
Comment (ry-itto) [reply app-ios/Sources/SponsorFeature/SponsorView.swift:180]: 👀 
I did not know that! 

> I want to know the best practice to fill the parent view 🤔

I want to know too... Let's find out together 👍🏼

**Review by ry-itto — APPROVED** [review]
LGTM! Thanks!!

> I feel scrolling is not smooth 🤔 (Images are too large??)

Let's check it in another issue/PR 👍🏼

**Comment by ry-itto** [comment]
@sudachi808 
There was a small and obvious conflict. I fix it to merge early 🙏🏼

**Pull request merged**